src/context/Theme.tsx

```diff
@@ -136,9 +140,14 @@ export function SettingsProvider({ children }: PropsWithChildren<{}>) {
     // check for the settings in the local storage
     const fromStorage = localStorage.getItem("settings")
     const values = fromStorage ? JSON.parse(fromStorage) : defaultProperties
+    // set the default for topBannerIsHidden based on the stored hash to avoid bad ux
+    // by shortly showing the banner and then hiding after client side effect
+    const currentTopBannerMessageHash = window.btoa(process.env.NEXT_PUBLIC_BANNER_MESSAGE)
```

Comment:
I don't think there's a point in encoding the string with `btoa`, moreover it's not a 100% safe and can throw an error on some utf characters.
I think you can just store the string as it is.